x-pack/solutions/observability/plugins/observability/public/pages/rule_details/components/header_actions.tsx

```diff
@@ -43,57 +71,142 @@ export function HeaderActions({
     onDeleteRule();
   };
 
-  return isRuleEditable ? (
-    <EuiFlexGroup direction="rowReverse" alignItems="flexStart">
-      <EuiFlexItem>
-        <EuiPopover
-          id="contextRuleEditMenu"
-          isOpen={isRuleEditPopoverOpen}
-          closePopover={togglePopover}
-          button={
-            <EuiButton
-              data-test-subj="actions"
-              disabled={isLoading}
-              fill
-              iconSide="right"
-              iconType="arrowDown"
-              onClick={togglePopover}
-            >
-              {i18n.translate('xpack.observability.ruleDetails.actionsButtonLabel', {
-                defaultMessage: 'Actions',
-              })}
-            </EuiButton>
-          }
-        >
-          <EuiFlexGroup direction="column" alignItems="flexStart" gutterSize="s">
-            <EuiButtonEmpty
-              data-test-subj="editRuleButton"
-              size="s"
-              iconType="pencil"
-              onClick={handleEditRule}
-            >
-              <EuiText size="s">
-                {i18n.translate('xpack.observability.ruleDetails.editRule', {
-                  defaultMessage: 'Edit rule',
-                })}
-              </EuiText>
-            </EuiButtonEmpty>
-            <EuiButtonEmpty
-              size="s"
-              iconType="trash"
-              color="danger"
-              onClick={handleRemoveRule}
-              data-test-subj="deleteRuleButton"
-            >
-              <EuiText size="s">
-                {i18n.translate('xpack.observability.ruleDetails.deleteRule', {
-                  defaultMessage: 'Delete rule',
+  const handleEnableRule = () => {
+    setIsRuleEditPopoverOpen(false);
+    enableRule({
+      id: ruleId,
+    });
+  };
+
+  const handleDisableRule = (untrack: boolean) => {
+    setIsRuleEditPopoverOpen(false);
+    onDisableModalClose();
+    disableRule({
+      id: ruleId,
+      untrack,
+    });
+  };
```

Comment:
Another opinionated comment.

```suggestion
```

Regarding `handleEditRule`, `handleRemoveRule`, `handleEnableRule`, `handleDisableRule`, these functions are only referenced once, further down where they're supplied as handlers to some components. Why not just put them there and leave the functions anonymous? The code will be closer to where it's called and will remove the mental load of the reader to go back up and figure out what each `handle` function is doing.

I realize that the first two `handle` functions were already there but I think this will simplify the component.